rust/noosphere-sphere/src/context.rs

```diff
@@ -254,20 +111,30 @@ where
         unimplemented!()
     }
 
-    /// Resolve the most recent version in the local history of the sphere
-    pub async fn head(&self) -> Result<Cid> {
-        let sphere_identity = self.identity();
-        self.db
-            .get_version(&self.sphere_identity)
-            .await?
-            .ok_or_else(|| anyhow!("No version found for {}", sphere_identity))
-    }
-
     /// The identity of the sphere
     pub fn identity(&self) -> &Did {
         &self.sphere_identity
     }
 
+    /// The identity of the counterpart sphere in use during this session, if
+    /// any; note that this will cause a request to be made to a gateway if no
+    /// handshake has yet occurred.
+    pub async fn counterpart_identity(&self) -> Result<Did> {
+        Ok(self.client().await?.session.sphere_identity.clone())
+    }
+
+    /// The identity of the counterpart sphere in use during this session, if
```

Comment:
nit: s/counterpart/gateway

Reply:
This is a bad mental model for implementation that lives on a sphere context. A gateway will make use of a sphere context, but it does not itself have a gateway to talk to.

This is confounded by the fact that sphere context's have a built-in client for the gateway API, so let's leave as is for now. But, we need to change this soon because it's a footgun when used within the gateway (or in the future, within a Geist).